DxDispatch/cmake/onnxruntime.cmake

```diff
@@ -0,0 +1,177 @@
+    endif()
+
+    # <PREFIX>_ONNXRUNTIME_TYPE
+    if(TARGET_WINDOWS AND TARGET_ARCH MATCHES "^X64|ARM64$")
+        set(default_type nuget)
+    elseif(TARGET_XBOX)
+        set(default_type local)
+    else()
+        set(default_type none)
+    endif()
+    set(${prefix}_ONNXRUNTIME_TYPE 
+        "${default_type}" 
+        CACHE STRING "ONNXRUNTIME dependency type"
+    )
+    set_property(CACHE ${prefix}_ONNXRUNTIME_TYPE PROPERTY STRINGS nuget local none)
+
+    # <PREFIX>_ONNXRUNTIME_NUGET_ID
+    set(${prefix}_ONNXRUNTIME_NUGET_ID
+        Microsoft.ML.OnnxRuntime.DirectML
+        CACHE STRING "ID of the ONNX Runtime NuGet package (TYPE == nuget)."
+    )
+
+    # <PREFIX>_ONNXRUNTIME_NUGET_VERSION
+    set(${prefix}_ONNXRUNTIME_NUGET_VERSION
+        1.11.0
+        CACHE STRING "Version of the ONNX Runtime NuGet package (TYPE == nuget)."
+    )
+
+    # <PREFIX>_ONNXRUNTIME_NUGET_HASH
+    set(${prefix}_ONNXRUNTIME_NUGET_HASH 
+        17ac3a5b3f7b4566aee9f29f39859f8ed26eb18cfbdfd2f06cb05ed230b484e2
+        CACHE STRING "SHA256 hash of the ONNX Runtime NuGet package (TYPE == nuget)."
+    )
+
+    # <PREFIX>_ONNXRUNTIME_LOCAL_PATH
+    set(${prefix}_ONNXRUNTIME_LOCAL_PATH
+        ""
+        CACHE STRING "Path to a local build of ONNX Runtime (TYPE == local)."
+    )
+endfunction()
+
+# -----------------------------------------------------------------------------
+# Init using a NuGet distribution.
+# -----------------------------------------------------------------------------
+function(init_onnxruntime_target_nuget target_name pkg_id pkg_version pkg_hash)
+    if(NOT (TARGET_WINDOWS AND TARGET_ARCH MATCHES "^X64|X86|ARM64$"))
+        message(FATAL_ERROR "ONNX Runtime NuGet isn't available on this platform")
+    endif()
+
+    set(content ${target_name}_content)
+    FetchContent_Declare(
+        ${content}
+        URL "https://www.nuget.org/api/v2/package/${pkg_id}/${pkg_version}"
+        URL_HASH SHA256=${pkg_hash}
+    )
+    FetchContent_MakeAvailable(${content})
+
+    target_include_directories(${target_name} INTERFACE "${${content}_SOURCE_DIR}/build/native/include")
+
+    set(bin_path "${${content}_SOURCE_DIR}/runtimes/win-${TARGET_ARCH}")
+    target_link_libraries(${target_name} INTERFACE "${bin_path}/native/onnxruntime.lib")
+    target_append_redist_file(${target_name} "${bin_path}/native/onnxruntime.dll")
+
+    set_property(TARGET ${target_name} PROPERTY DX_COMPONENT_CONFIG "NuGet (${pkg_id}.${pkg_version})")
+endfunction()
+
+# -----------------------------------------------------------------------------
+# Init using a local build.
+# -----------------------------------------------------------------------------
+function(init_onnxruntime_target_local target_name local_path)
+    if(NOT IS_DIRECTORY "${local_path}")
+        message(FATAL_ERROR "'${local_path}' is not a directory. You must set DXD_ONNXRUNTIME_LOCAL_PATH to a directory containing pre-built DirectML.")
```

Comment:
>DXD [](http://example.com/codeflow?start=78&length=3)

DML? (probably not a typo, but checking)

Reply:
Intentional (short for dxdispatch), but of dubious value. The idea was to avoid having a generic CMake cache variable name in case this project was ever consumed by another project, but in reality having multiple versions of these components would be unlikely and probably a bad idea.